src/main/modules/app-insights/index.ts

```diff
@@ -1,35 +1,24 @@
 import * as config from 'config'
 import * as appInsights from 'applicationinsights'
+import * as telemetryProcessors from './telemetryProcessors'
+import { LoggerInstance } from 'winston'
+import { Logger } from '@hmcts/nodejs-logging'
 
-const fileRegexp = new RegExp('(\\..{2,5}$)')
-const uuidRegexp = new RegExp('[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}')
-
-function hideUuidInUrlIfNotStaticFile (url: string): string {
-  if (!fileRegexp.test(url)) {
-    return url.replace(uuidRegexp, '{uuid}')
-  } else {
-    return url
-  }
-}
+const logger: LoggerInstance = Logger.getLogger('customEventTracker')
 
 export class AppInsights {
   static enable () {
 
-    appInsights.setup(config.get<string>('secrets.cmc.AppInsightsInstrumentationKey'))
+    const aiInstrumentationKey = config.get<string>('secrets.cmc.AppInsightsInstrumentationKey')
+    appInsights.setup(aiInstrumentationKey)
       .setAutoCollectConsole(true, true)
     appInsights.defaultClient.context.tags[appInsights.defaultClient.context.keys.cloudRole] = config.get<string>('appInsights.roleName')
 
-    appInsights.defaultClient.addTelemetryProcessor(function (envelope, contextObjects) {
-      // hide UUID's in operationName URL's that are not static files, so they can be aggregated properly
-      if (envelope.tags) {
-        if (envelope.tags['ai.operation.name']) {
-          envelope.tags['ai.operation.name'] = hideUuidInUrlIfNotStaticFile(envelope.tags['ai.operation.name'])
-        }
-      }
+    appInsights.defaultClient.addTelemetryProcessor(telemetryProcessors.operationNameUUIDHider())
 
-      // always send
-      return true
-    })
+    if (aiInstrumentationKey === 'STDOUT') {
```

Comment:
Should we have a test around this branch?

Reply:
There were previously zero tests for this file; I refactored the telemetry bits out purely for testability. Not sure how I would go about testing this in TypeScript. If you know how, please share! As it stands, it's already improved coverage

Reply:
Have tried using `mock-require` but couldn't get it working. I don't feel it's worth spending hours trying to unit test this.